packages/client/src/runtime/core/engines/data-proxy/utils/request.ts

```diff
@@ -142,3 +146,45 @@ async function nodeFetch(url: string, options: RequestOptions = {}): Promise<Req
 
 // trick to obfuscate require from bundlers, useful for Vercel Edge
 const include = typeof require !== 'undefined' ? require : () => {}
+
+export class NodeHeaders {
+  readonly #headers = new Map<string, string>()
```

Comment:
Back in the day there were problems with private fields: https://prisma-company.slack.com/archives/CUXLS0Z6K/p1675164034311419.
We have no explicit decision on whether or not it's ok to break `"target": "ES3"` people, but maybe we should now.

Reply:
It looks like the minimum would be `ES2022` for this.
I think to be safe, we would need to
- Check the impact of this
- Check it fails and document what to do?
- See what's the default in the important cases, for example is it es2021 by default for [Cloudflare Workers](https://github.com/cloudflare/workers-sdk/blob/624846d574746d9eca575ef0d24f32e20605980c/packages/wrangler/templates/tsconfig.init.json#L14)?
- Mention this in the docs
- Check ecosystem-tests
- Check our examples
- Anything else?

Reply:
Let me remove the private property for the time being, this will go away soon anyways.